src/Root.tsx

```diff
@@ -1,10 +1,18 @@
 import React, { Component } from 'react';
-import { createStore } from 'redux';
+import { combineReducers, createStore, applyMiddleware } from 'redux';
+import createSagaMiddleware from 'redux-saga';
 import { Provider } from 'react-redux';
-import counterReducer from './reducers';
+import rootReducer from './reducers';
+import { counterSaga } from './sagas';
 import App from './containers/App';
 
-const store = createStore(counterReducer);
+const sagaMiddleware = createSagaMiddleware();
```

Comment:
All of this store creation should be in another file so it's easier to find and modify in the future.